File: corpus/src/javascript/normalisation.js
// Input written the way a person writes it and prettier does not: wrong spacing
// around operators and delimiters, padding inside brackets, wrong indentation,
// runs of spaces before a trailing comment.
function weird( a , b ){
    const result=a+b* 2;
    return result;
}

const emptyCall = f( );
const emptyArray = [ ];
const emptyObject = { };

const padded=[ 1,2,3 ];
const tight={ x:1,y:2 };
const spaced = { a : 1 , b : 2 };

const redundant = ((1 + 2));

if ( x ){ y=1 }else{ y=2 }

const trailingSpaces = 1       // lots of spaces before this comment

function indented() {
		const n = 0;
		if (n === 0) {
				n = 1;
		}
		return n;
}

File: corpus/src/kotlin/normalisation.kt
// Dirty spacing: what ktfmt rewrites, not what it breaks.

fun  normalise( a : Int ,b:Int ):Int{
val result=a + b*  2
return(result)
}

fun empty( ) { // empty parameter list written with a space
val xs = listOf( ) // empty call with a space inside
val ys = mapOf( )
val zs = setOf(  )
}

fun padded( ) {
val x = listOf( 1,2,3 )
if ( x != null ) { println(x) } else { println(0) }
}

val trailingSpaces = 1       // lots of spaces before this comment

fun indented() {
        val n = 0
        if(n==0){
                println(n)
        }
}

val semis = 1; val two = 2; val three = 3
